llvm/lib/Target/AMDGPU/SOPInstructions.td

```diff
@@ -213,6 +213,11 @@ let isMoveImm = 1 in {
   } // End Uses = [SCC]
 } // End isMoveImm = 1
 
+// Variant of S_MOV_B32 used for reading volatile source values like
+// SRC_POPS_EXITING_WAVE_ID.
+let mayLoad = 1, mayStore = 1, maybeAtomic = 0 in
+def S_MOV_B32_loadstore : SOP1_32 <"s_mov_b32">;
```

Comment:
I have not yet managed to prevent TableGen from complaining about selecting an IntrInaccessibleMemOnly intrinsic to an instruction without mayLoad/mayStore. The flag inference stuff in CodeGenDAGPatterns::InferFromPattern is just too complex.

Reply:
Can we just delete it?

Reply:
I'd rather avoid the pseudo. I'd probably prefer to just manually select over having it. But really that flag inference stuff in TableGen is garbage. We at minimum need to fix the properties to exactly match the attributes

Reply:
OK, I updated the patch to select it in C++ (and rebased).